ngraph/core/src/function.cpp

```diff
@@ -449,7 +479,32 @@ bool ov::Function::evaluate(const HostTensorVector& output_tensors,
     for (const auto& m_sink : m_sinks) {
         outputs.push_back(m_sink);
     }
-    ngraph::evaluate_nodes(value_map, output_tensor_map, outputs, evaluation_context);
+    // evaluate nodes
+    OPENVINO_SUPPRESS_DEPRECATED_START
+    ngraph::Evaluator<ov::runtime::Tensor::Ptr> evaluator({}, value_map);
+    evaluator.set_univeral_handler(
+        [&output_tensor_map,
+         &evaluation_context](Node* node, const ov::runtime::TensorVector& input_tensors) -> ov::runtime::TensorVector {
+            ov::runtime::TensorVector output_tensors;
+            for (const auto& v : node->outputs()) {
+                auto it = output_tensor_map.find(v);
+                if (it == output_tensor_map.end()) {
+                    auto c = make_shared<ov::runtime::Tensor>(v.get_element_type(), v.get_shape());
```

Comment:
You do not need `shared_ptr` to store `ov::runtime::Tensor`

Reply:
The old evaluate methods work with vector of pointers to tensor. In new API I use the same approach.

Reply:
Yes, I will change the behavior